lib/grandpa/message_handler.go

```diff
@@ -573,6 +609,18 @@ func (s *Service) VerifyBlockJustification(hash common.Hash, justification []byt
 	return nil
 }
 
+func verifyBlockHashAgainstBlockNumber(bs BlockState, hash common.Hash, number uint) error {
+	header, err := bs.GetHeader(hash)
+	if err != nil {
+		return fmt.Errorf("could not get header from block hash: %w", err)
```

Comment:
is this error is returned, we likely haven't synced this block yet, and should add the message to the `tracker` for processing later. otherwise this will cause finality to get stuck if syncing is lagging

Reply:
addressed